bench/tx-generator/src/Cardano/Benchmarking/Script/Core.hs

```diff
@@ -632,6 +643,37 @@ createChangeGeneric submitMode createCoins addressMsg value count = do
   chunkList _ [] = []
   chunkList n xs = as : chunkList n bs where (as,bs) = splitAt n xs
 
+spendAutoScript :: SubmitMode -> FilePath -> ThreadName -> NumberOfTxs -> TPSRate -> ActionM ()
+spendAutoScript submitMode scriptFile threadName txCount tps = do
+  protocolParameters <- queryProtocolParameters
+  budget <- case protocolParamMaxTxExUnits protocolParameters of
+    Nothing -> throwE $ ApiError "cannot determine protocolParamMaxTxExUnits"
+    Just b -> return b
+  script <- liftIO $ readScript scriptFile
+  let
+    costFn :: Integer -> Either String Bool
+    costFn n = case preExecuteScript protocolParameters script (ScriptDataNumber 0) (ScriptDataNumber $ n + 1000000) of
+      Left err -> Left err
+      Right use -> Right $ (executionSteps use <= executionSteps budget) && (executionMemory use <= executionMemory budget)
+
+  redeemer <- case startSearch costFn 0 100000 of
+    Left err -> throwE $ ApiError $ "cannot find fitting redeemer :" ++ err
+    Right x -> return $ ScriptDataNumber $ x + 1000000
```

Comment:
We're hardcoding a number of things here and there.

It'd be nice to have named top-level variables, instead.

Reply:
Yes..And 1_000_000 is really hard coded !
The 1_000_000 must match the hard coded value in the Plutus script.
The 100_000 is just an arbitrary large number.

Reply:
The way to do this is to introduce named "packages", which come from outside:
```
{
   scriptFile: "foo.json",
   scriptConstantA: 1000000

Reply:
..and then you suddenly can reformulate script processing around well-defined ADTs..

Reply:
I think we should leave it as it is for now.
The proposed indirection just moves the magic number from inside the Haskell code to
some JSON file and causes a 20 times code-blow-up.
This number is strictly non-serviceable for now.
It must match the number that is hard coded in the plutus-example/loop.hs
This may be changed later (if ever needed).
(It is also possible to handle this just in loop.hs.)